Bugfix rspamd config errors

Jumper78 opened the pull request:
## What type of PR?

This **bug-fix** addresses configuration errors that show up in the rspamd module.

## What does this PR do?

 When I performed a configdump in the antispam module (e.g., by executing `docker exec mailu-antispam-1 rspamadm configdump mx_check`), I received the following error messages:

```
unknown worker attribute: type; worker type: normal
unknown worker attribute: type; worker type: controller
cannot add dependency from OLETOOLS_MACRO_MRAPTOR on FORCE_ACTION_ANTIVIRUS_FLAGGED: invalid symbol types
cannot add dependency from OLETOOLS_MACRO_SUSPICIOUS on FORCE_ACTION_ANTIVIRUS_FLAGGED: invalid symbol types
cannot add dependency from OLETOOLS_FAIL on FORCE_ACTION_ANTIVIRUS_FAILED: invalid symbol types
```

### Related issue(s)
- the errors were shown as well in issue #3262.
- the bug-fix should also close some of the errors in discussion #3263

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

Jumper78 committed:
corrected the syntax of the worker files

Jumper78 committed:
added mission semicolon at the end of the server= line

Jumper78 committed:
cleaned up not needed modifications

Jumper78 committed:
Revert "cleaned up not needed modifications"

This reverts commit dddbc94917101cf632f521635c8fb1440dc9c967.

Jumper78 committed:
added honor_action lines to force_actions.conf

Jumper78 committed:
corrected honor_action lines

Jumper78 committed:
cleaned up unnecessary changes

Jumper78 committed:
Create 3784.bugfix

Jumper78 commented:
Short description what I changed and why I think that is correct:
- in `core/rspamd/conf/external_services.conf` there was a semi colon missing
- `core/rspamd/conf/force_actions.conf` caused the "cannot add dependency" errors
  - the `honor_action` lines make sure that these force actions do not overwrite a higher ranking action
  - with the `honor_action` lines there are no more dependency error messages
- the attribute `type` is not needed anymore in the worker configurations
  - it causes the described errors
  - the attribute is already set in a higher ranking config file (https://github.com/rspamd/rspamd/blob/22b1d712b36654514db359dbebe1d096d828fb05/conf/rspamd.conf#L67)

Jumper78 commented:
bors try

nextgens commented:
bors try

Jumper78 commented:
Unfortunately, the tests are right. I still have a problem with my bugfix.
I confirmed on my private test-server that Mailu now accepts mails with `2003x32_word_msgbox_stomped_fakecode.doc` as attachment ... and that is not good.

nextgens commented:
FYI the example given at https://rspamd.com/doc/modules/external_services.html#oletools-default-mode is also missing the semicolon

nextgens commented:
@Jumper78 can you get to the bottom of this please?

Jumper78 committed:
Merge branch 'Mailu:master' into bugfix-rspamd-config-errors

Jumper78 commented:
@nextgens , I will try my best to find a solution.
Could you do me a favour? Could you start another `bors try`? The old logs are gone and I would like to compare the test results with the results on my private test server.

Jumper78 commented:
Just a quick comment about the current status of my analysis. I changed three things in this PR:
- I deleted the lines with `type = {...}` --> these changes do not destroy the functionality
- I added a semicolon --> this change also does not destroy the functionality
- I added `honor_action` lines to `force_actions.conf` --> those are the ones leading to the file `2003x32_word_msgbox_stomped_fakecode.doc` not being blocked by the mail server

I also checked the functionality of the force actions. Unfortunately, the force actions based on the OLE-Tools seem to not work properly in the current master tree. Mails with `2003x32_word_msgbox_stomped_fakecode.doc` as attachment are rejected with `554 5.7.1 Spam message rejected\r\n` while they should be rejected with the comment `Rejected (dangerous/malicious code detected)`. So, I am still convinced that the force actions are not work as intended and that should be fixed.
I already have a solution where the force actions are working, but there are still error-messages that should not be there. I will continue working on it.

Jumper78 committed:
Merge branch 'Mailu:master' into bugfix-rspamd-config-errors

Jumper78 committed:
Update force_actions.conf

taken out the honor_action lines;
separted the oletools parts from the clam virus parts;
substituted the composites parts by their original expressions;

Jumper78 committed:
Merge pull request #14 from Jumper78/bugfix-rspamd-config-errors-temp

Update force_actions.conf

Jumper78 commented:
Unfortunately, I am not able to get rid of all of the error messages. Two errors remain when I run `docker exec mailu-antispam-1 rspamadm configdump mx_check`, I still get:
`cannot add dependency from OLETOOLS_FAIL on FORCE_ACTION_ANTIVIRUS_FAILED: invalid symbol types`
`cannot add dependency from OLETOOLS_A on FORCE_ACTION_OLETOOLS_DANGEROUS: invalid symbol types`
But with the current solution the force_action rule OLETOOLS_DANGEROUS works properly and E-Mails detected by the rules are rejected due to force_actions and not only due to the spam score.
@nextgens, could you retrigger `bors try`?

nextgens reviewed (approved):
LGTM
